pkg/kubectl/cmd/drain.go

```diff
@@ -619,7 +620,7 @@ func (o *DrainOptions) evictPods(pods []corev1.Pod, policyGroupVersion string, g
 			if doneCount == len(pods) {
 				return nil
 			}
-		case <-time.After(globalTimeout):
+		case <-globalTimeoutCh:
```

Comment:
Does it make any difference with the previous implementation?

Reply:
Yes, in the previous implementation it's basically a timeout per pod.

Run the following code locally and notice it never ends because the timer gets reset on each for loop iteration: https://play.golang.org/p/uq1I8nLx5Af

Reply:
> The length of time to wait before giving up, zero means infinite

Umm, yes. The timeout here refreshes every time when a pod is successfully evicted which is inconsistent with the docs.